power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -350,6 +354,79 @@ template <bool sym> class YBus {
         admittance_ = std::make_shared<ComplexTensorVector<sym> const>(std::move(admittance));
     }
 
+    // void increments_to_entries(std::shared_ptr<IdxVector> affected_entries) {
+    IdxVector increments_to_entries() {
+        // construct affected entries
+        IdxVector affected_entries = {};
+        auto const& y_bus_pos_in_entries = y_bus_struct_->y_bus_pos_in_entries;
+        auto process_params = [&affected_entries, &y_bus_pos_in_entries, this](auto params_to_change, bool is_branch) {
+            for (auto param_to_change : params_to_change) {
+                Idx id_x{0};
+                Idx id_y{0};
+                if (is_branch) {
+                    id_x = math_topology_->branch_bus_idx[param_to_change][0];
+                    id_y = math_topology_->branch_bus_idx[param_to_change][1];
+                } else {
+                    id_x = math_topology_->shunts_per_bus.get_group(param_to_change);
+                    id_y = math_topology_->shunts_per_bus.get_group(param_to_change);
+                }
+                auto it =
+                    std::ranges::find(y_bus_pos_in_entries.begin(), y_bus_pos_in_entries.end(), MatrixPos{id_x, id_y});
+                if (it != y_bus_pos_in_entries.end()) {
+                    affected_entries.push_back(std::distance(y_bus_pos_in_entries.begin(), it));
+                }
+            }
+        };
+
+        process_params(math_model_param_incrmt_->branch_param_to_change, true);
+        process_params(math_model_param_incrmt_->shunt_param_to_change, false);
+        return affected_entries;
+    }
+
+    void update_admittance_increment(std::shared_ptr<MathModelParamIncrement<sym> const> const& math_model_param_incrmt,
+                                     bool is_decrement = false) {
+        const double mode = is_decrement ? -1.0 : 1.0;
+        if (!is_decrement) {
```

Comment:
i still don't like the `increment` and `decrement` names. technically, you don't add or subtract values from a common state object (which the names suggest), but invalidate entries and reconstruct them from scratch.

In addition, the update inverse that is stored in the `main_model` already contains the update data that is required to end with the original ybus contributions. When this piece of code is reached, regardless of whether an update or an inverse update was applied, there should be no difference between the two cases.

The only exception could be the case where `U o U^-1` is applied, which amounts to just the identity operation. Is this code about that? If so, i'm not sure whether this is the correct location...

Reply:
I did some digging in `main::core` and it seems there's some re-works coming in that direction, so it'll soon make sense (good news: we don't have to wait until April)

Reply:
To be discussed face-to-face;